src/pages/ConnectWallet/ConnectWallet.tsx

```diff
@@ -14,79 +15,79 @@ import OrbsStatic from 'assets/orbs-static.png'
 import { FoxIcon } from 'components/Icons/FoxIcon'
 import { Page } from 'components/Layout/Page'
 import { RawText, Text } from 'components/Text'
+import { KeyManager, SUPPORTED_WALLETS } from 'context/WalletProvider/config'
 import { ActionTypes, InitialState, WalletActions } from 'context/WalletProvider/WalletProvider'
 import { useQuery } from 'hooks/useQuery/useQuery'
 import { colors } from 'theme/colors'
 
-import { KeyManager, SUPPORTED_WALLETS } from '../../context/WalletProvider/config'
-
 type WalletProps = {
   state: InitialState
   dispatch: Dispatch<ActionTypes>
 }
 
+async function connectCypressWallet(
+  keyring: Keyring,
+  dispatch: Dispatch<ActionTypes>,
+  walletId: string,
+  walletPassword: string
+) {
+  const adapter = SUPPORTED_WALLETS[KeyManager.Native].adapter.useKeyring(keyring)
+  if (adapter) {
+    const wallet = await adapter.pairDevice(walletId)
+    await wallet.initialize()
+  }
+
+  const vaultIds = await Vault.list()
+  for (let index = 0; index < vaultIds.length; index++) {
```

Comment:
We can tighten this up a touch with a `for...of` loop:

```ts
for (const deviceId of vaultIds) {
    if (deviceId !== walletId) {for (const deviceId of vaultIds) {
...
```

Reply:
No `for loop` anymore in current implementation.